File: 283.移动零.java
/*
 * @lc app=leetcode.cn id=283 lang=java
 *
 * [283] 移动零
 */

// @lc code=start
class Solution {
    public void moveZeroes(int[] nums) {
        // 双指针，把非零数值填充到前面，剩下的填充0
        int slow = 0, fast = 0, n = nums.length;

        while(fast < n) {
            if (nums[fast] != 0) {
                nums[slow++] = nums[fast++];
            } else {
                fast++;
            }
        }
        while(slow < n) {
            nums[slow++] = 0;
        }
    }
}
// @lc code=end

File: 283.移动零.js
/*
 * @lc app=leetcode.cn id=283 lang=javascript
 *
 * [283] 移动零
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {

    let l = 0, r = 0;

    while(r < nums.length) {
        if (nums[r] === 0) {
            r++;
        } else {
            nums[l] = nums[r];
            l++;
            r++;
        }
    }

    while(l < nums.length) {
        nums[l] = 0;
        l++;
    }

};
// @lc code=end

